webrender/src/frame_builder.rs

```diff
@@ -1778,6 +1779,10 @@ impl FrameBuilder {
                 &self.clip_store,
                 RenderPassIndex(pass_index),
             );
+
+            if let RenderPassKind::OffScreen { ref texture_cache, .. } = pass.kind {
```

Comment:
hmm, so any non-main-FBO tasks are considered for `has_texture_cache_tasks`? If the problem is only about skipping the tasks that are required for future rendering, we need a distinction between re-usable and non-reusable tasks, and only checks for the former here

Reply:
It checks for the ```texture_cache``` target being non-empty. So this excludes temporary render tasks.